internal/provider/gce/environ_broker.go

```diff
@@ -203,18 +194,30 @@ func (env *environ) startInstance(
 		allocatePublicIP = *args.Constraints.AllocatePublicIP
 	}
 
-	vpcLink, subnetURL, err := env.getInstanceSubnet(ctx)
+	var nics []*computepb.NetworkInterface
+	vpcLink, subnets, err := env.subnetsForInstance(ctx, args)
 	if err != nil {
-		return nil, google.HandleCredentialError(errors.Trace(err), ctx)
+		return nil, environs.ZoneIndependentError(err)
 	}
-	nic := &computepb.NetworkInterface{
-		Network:    &vpcLink,
-		Subnetwork: subnetURL,
+	if len(subnets) == 0 {
+		nics = []*computepb.NetworkInterface{{
+			Network: vpcLink,
+		}}
+	} else {
+		for _, subnet := range subnets {
+			if err := isSubnetReady(subnet); err != nil {
+				return nil, environs.ZoneIndependentError(err)
+			}
+			nics = append(nics, &computepb.NetworkInterface{
```

Comment:
I thought you mentioned that we didn't do multi-nic for single VPC, or was it the other way around?